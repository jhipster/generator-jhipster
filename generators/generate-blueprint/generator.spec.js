/**
 * Copyright 2013-2023 the original author or authors from the JHipster project.
 *
 * This file is part of the JHipster project, see https://www.jhipster.tech/
 * for more information.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { basename, dirname, join } from 'path';
import { fileURLToPath } from 'url';
import { before, it, describe, expect } from 'esmocha';
import lodash from 'lodash';

import { defaultHelpers as helpers } from '../../test/support/index.js';
import { shouldSupportFeatures, testBlueprintSupport } from '../../test/support/tests.js';
import Generator from './index.js';

const { snakeCase } = lodash;

const __filename = fileURLToPath(import.meta.url);
const __dirname = dirname(__filename);

const generatorPath = join(__dirname, 'index.js');
const generator = basename(__dirname);

const mockedGenerators = ['jhipster:init'];

describe(`generator - ${generator}`, () => {
  it('generator-list constant matches folder name', async () => {
    const generatorList = await import('../generator-list.js');
    await expect(generatorList[`GENERATOR_${snakeCase(generator).toUpperCase()}`]).toBe(generator);
  });
  shouldSupportFeatures(Generator);
  describe('blueprint support', () => testBlueprintSupport(generator));

  describe('with', () => {
    describe('default config', () => {
      let runResult;
      before(async () => {
        runResult = await helpers.run(generatorPath).withJHipsterConfig().withMockedGenerators(mockedGenerators);
      });
      it('should compose with init generator', () => {
        expect(runResult.mockedGenerators['jhipster:init'].calledOnce).toBe(true);
      });
      it('should write files and match snapshot', () => {
        expect(runResult.getStateSnapshot()).toMatchSnapshot();
      });
    });
    describe('all option', () => {
      let runResult;
      before(async () => {
        runResult = await helpers.run(generatorPath).withOptions({ allGenerators: true }).withMockedGenerators(mockedGenerators);
      });
      it('should compose with init generator', () => {
        expect(runResult.mockedGenerators['jhipster:init'].calledOnce).toBe(true);
      });
      it('should match snapshot', () => {
        expect(runResult.getStateSnapshot()).toMatchSnapshot();
      });
    });
    describe('local-blueprint option', () => {
      let runResult;
      before(async () => {
        runResult = await helpers.run(generatorPath).withOptions({ localBlueprint: true }).withMockedGenerators(mockedGenerators);
      });
      it('should not compose with init generator', () => {
        expect(runResult.mockedGenerators['jhipster:init'].calledOnce).toBe(false);
      });
      it('should match snapshot', () => {
        expect(runResult.getStateSnapshot()).toMatchInlineSnapshot(`
{
  ".yo-rc.json": {
    "stateCleared": "modified",
  },
}
`);
      });
    });
    describe('local-blueprint option and app generator', () => {
      let runResult;
      before(async () => {
        runResult = await helpers
          .run(generatorPath)
          .withOptions({ localBlueprint: true, subGenerators: ['app'], allPriorities: true })
          .withMockedGenerators(mockedGenerators);
      });
      it('should not compose with init generator', () => {
        expect(runResult.mockedGenerators['jhipster:init'].calledOnce).toBe(false);
      });
      it('should write java files with gradle build tool and match snapshot', () => {
        expect(runResult.getStateSnapshot()).toMatchInlineSnapshot(`
{
  ".blueprint/app/command.mjs": {
    "stateCleared": "modified",
  },
  ".blueprint/app/generator.mjs": {
    "stateCleared": "modified",
  },
  ".blueprint/app/index.mjs": {
    "stateCleared": "modified",
  },
  ".blueprint/app/templates/template-file-app.ejs": {
    "stateCleared": "modified",
  },
  ".yo-rc.json": {
    "stateCleared": "modified",
  },
}
`);
      });
    });
  });
});
