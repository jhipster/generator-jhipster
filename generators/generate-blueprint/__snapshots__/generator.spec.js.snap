// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generator - generate-blueprint with all option should match snapshot 1`] = `
{
  ".blueprint/cli/commands.mjs": {
    "stateCleared": "modified",
  },
  ".blueprint/generate-sample/command.mjs": {
    "stateCleared": "modified",
  },
  ".blueprint/generate-sample/generator.mjs": {
    "stateCleared": "modified",
  },
  ".blueprint/generate-sample/index.mjs": {
    "stateCleared": "modified",
  },
  ".blueprint/generate-sample/templates/samples/sample.jdl": {
    "stateCleared": "modified",
  },
  ".eslintrc.json": {
    "stateCleared": "modified",
  },
  ".github/workflows/generator.yml": {
    "stateCleared": "modified",
  },
  ".yo-rc.json": {
    "stateCleared": "modified",
  },
  "README.md": {
    "stateCleared": "modified",
  },
  "cli/cli.cjs": {
    "stateCleared": "modified",
  },
  "generators/angular/command.mjs": {
    "stateCleared": "modified",
  },
  "generators/angular/generator.mjs": {
    "stateCleared": "modified",
  },
  "generators/angular/generator.spec.mjs": {
    "stateCleared": "modified",
  },
  "generators/angular/index.mjs": {
    "stateCleared": "modified",
  },
  "generators/angular/templates/template-file-angular.ejs": {
    "stateCleared": "modified",
  },
  "generators/app/command.mjs": {
    "stateCleared": "modified",
  },
  "generators/app/generator.mjs": {
    "stateCleared": "modified",
  },
  "generators/app/generator.spec.mjs": {
    "stateCleared": "modified",
  },
  "generators/app/index.mjs": {
    "stateCleared": "modified",
  },
  "generators/app/templates/template-file-app.ejs": {
    "stateCleared": "modified",
  },
  "generators/base-application/command.mjs": {
    "stateCleared": "modified",
  },
  "generators/base-application/generator.mjs": {
    "stateCleared": "modified",
  },
  "generators/base-application/generator.spec.mjs": {
    "stateCleared": "modified",
  },
  "generators/base-application/index.mjs": {
    "stateCleared": "modified",
  },
  "generators/base-application/templates/template-file-base-application.ejs": {
    "stateCleared": "modified",
  },
  "generators/base-core/command.mjs": {
    "stateCleared": "modified",
  },
  "generators/base-core/generator.mjs": {
    "stateCleared": "modified",
  },
  "generators/base-core/generator.spec.mjs": {
    "stateCleared": "modified",
  },
  "generators/base-core/index.mjs": {
    "stateCleared": "modified",
  },
  "generators/base-core/templates/template-file-base-core.ejs": {
    "stateCleared": "modified",
  },
  "generators/base-entity-changes/command.mjs": {
    "stateCleared": "modified",
  },
  "generators/base-entity-changes/generator.mjs": {
    "stateCleared": "modified",
  },
  "generators/base-entity-changes/generator.spec.mjs": {
    "stateCleared": "modified",
  },
  "generators/base-entity-changes/index.mjs": {
    "stateCleared": "modified",
  },
  "generators/base-entity-changes/templates/template-file-base-entity-changes.ejs": {
    "stateCleared": "modified",
  },
  "generators/base-workspaces/command.mjs": {
    "stateCleared": "modified",
  },
  "generators/base-workspaces/generator.mjs": {
    "stateCleared": "modified",
  },
  "generators/base-workspaces/generator.spec.mjs": {
    "stateCleared": "modified",
  },
  "generators/base-workspaces/index.mjs": {
    "stateCleared": "modified",
  },
  "generators/base-workspaces/templates/template-file-base-workspaces.ejs": {
    "stateCleared": "modified",
  },
  "generators/base/command.mjs": {
    "stateCleared": "modified",
  },
  "generators/base/generator.mjs": {
    "stateCleared": "modified",
  },
  "generators/base/generator.spec.mjs": {
    "stateCleared": "modified",
  },
  "generators/base/index.mjs": {
    "stateCleared": "modified",
  },
  "generators/base/templates/template-file-base.ejs": {
    "stateCleared": "modified",
  },
  "generators/bootstrap-application-base/command.mjs": {
    "stateCleared": "modified",
  },
  "generators/bootstrap-application-base/generator.mjs": {
    "stateCleared": "modified",
  },
  "generators/bootstrap-application-base/generator.spec.mjs": {
    "stateCleared": "modified",
  },
  "generators/bootstrap-application-base/index.mjs": {
    "stateCleared": "modified",
  },
  "generators/bootstrap-application-base/templates/template-file-bootstrap-application-base.ejs": {
    "stateCleared": "modified",
  },
  "generators/bootstrap-application-client/command.mjs": {
    "stateCleared": "modified",
  },
  "generators/bootstrap-application-client/generator.mjs": {
    "stateCleared": "modified",
  },
  "generators/bootstrap-application-client/generator.spec.mjs": {
    "stateCleared": "modified",
  },
  "generators/bootstrap-application-client/index.mjs": {
    "stateCleared": "modified",
  },
  "generators/bootstrap-application-client/templates/template-file-bootstrap-application-client.ejs": {
    "stateCleared": "modified",
  },
  "generators/bootstrap-application-server/command.mjs": {
    "stateCleared": "modified",
  },
  "generators/bootstrap-application-server/generator.mjs": {
    "stateCleared": "modified",
  },
  "generators/bootstrap-application-server/generator.spec.mjs": {
    "stateCleared": "modified",
  },
  "generators/bootstrap-application-server/index.mjs": {
    "stateCleared": "modified",
  },
  "generators/bootstrap-application-server/templates/template-file-bootstrap-application-server.ejs": {
    "stateCleared": "modified",
  },
  "generators/bootstrap-application/command.mjs": {
    "stateCleared": "modified",
  },
  "generators/bootstrap-application/generator.mjs": {
    "stateCleared": "modified",
  },
  "generators/bootstrap-application/generator.spec.mjs": {
    "stateCleared": "modified",
  },
  "generators/bootstrap-application/index.mjs": {
    "stateCleared": "modified",
  },
  "generators/bootstrap-application/templates/template-file-bootstrap-application.ejs": {
    "stateCleared": "modified",
  },
  "generators/bootstrap-workspaces/command.mjs": {
    "stateCleared": "modified",
  },
  "generators/bootstrap-workspaces/generator.mjs": {
    "stateCleared": "modified",
  },
  "generators/bootstrap-workspaces/generator.spec.mjs": {
    "stateCleared": "modified",
  },
  "generators/bootstrap-workspaces/index.mjs": {
    "stateCleared": "modified",
  },
  "generators/bootstrap-workspaces/templates/template-file-bootstrap-workspaces.ejs": {
    "stateCleared": "modified",
  },
  "generators/bootstrap/command.mjs": {
    "stateCleared": "modified",
  },
  "generators/bootstrap/generator.mjs": {
    "stateCleared": "modified",
  },
  "generators/bootstrap/generator.spec.mjs": {
    "stateCleared": "modified",
  },
  "generators/bootstrap/index.mjs": {
    "stateCleared": "modified",
  },
  "generators/bootstrap/templates/template-file-bootstrap.ejs": {
    "stateCleared": "modified",
  },
  "generators/ci-cd/command.mjs": {
    "stateCleared": "modified",
  },
  "generators/ci-cd/generator.mjs": {
    "stateCleared": "modified",
  },
  "generators/ci-cd/generator.spec.mjs": {
    "stateCleared": "modified",
  },
  "generators/ci-cd/index.mjs": {
    "stateCleared": "modified",
  },
  "generators/ci-cd/templates/template-file-ci-cd.ejs": {
    "stateCleared": "modified",
  },
  "generators/client/command.mjs": {
    "stateCleared": "modified",
  },
  "generators/client/generator.mjs": {
    "stateCleared": "modified",
  },
  "generators/client/generator.spec.mjs": {
    "stateCleared": "modified",
  },
  "generators/client/index.mjs": {
    "stateCleared": "modified",
  },
  "generators/client/templates/template-file-client.ejs": {
    "stateCleared": "modified",
  },
  "generators/common/command.mjs": {
    "stateCleared": "modified",
  },
  "generators/common/generator.mjs": {
    "stateCleared": "modified",
  },
  "generators/common/generator.spec.mjs": {
    "stateCleared": "modified",
  },
  "generators/common/index.mjs": {
    "stateCleared": "modified",
  },
  "generators/common/templates/template-file-common.ejs": {
    "stateCleared": "modified",
  },
  "generators/cucumber/command.mjs": {
    "stateCleared": "modified",
  },
  "generators/cucumber/generator.mjs": {
    "stateCleared": "modified",
  },
  "generators/cucumber/generator.spec.mjs": {
    "stateCleared": "modified",
  },
  "generators/cucumber/index.mjs": {
    "stateCleared": "modified",
  },
  "generators/cucumber/templates/template-file-cucumber.ejs": {
    "stateCleared": "modified",
  },
  "generators/cypress/command.mjs": {
    "stateCleared": "modified",
  },
  "generators/cypress/generator.mjs": {
    "stateCleared": "modified",
  },
  "generators/cypress/generator.spec.mjs": {
    "stateCleared": "modified",
  },
  "generators/cypress/index.mjs": {
    "stateCleared": "modified",
  },
  "generators/cypress/templates/template-file-cypress.ejs": {
    "stateCleared": "modified",
  },
  "generators/docker-compose/command.mjs": {
    "stateCleared": "modified",
  },
  "generators/docker-compose/generator.mjs": {
    "stateCleared": "modified",
  },
  "generators/docker-compose/generator.spec.mjs": {
    "stateCleared": "modified",
  },
  "generators/docker-compose/index.mjs": {
    "stateCleared": "modified",
  },
  "generators/docker-compose/templates/template-file-docker-compose.ejs": {
    "stateCleared": "modified",
  },
  "generators/docker/command.mjs": {
    "stateCleared": "modified",
  },
  "generators/docker/generator.mjs": {
    "stateCleared": "modified",
  },
  "generators/docker/generator.spec.mjs": {
    "stateCleared": "modified",
  },
  "generators/docker/index.mjs": {
    "stateCleared": "modified",
  },
  "generators/docker/templates/template-file-docker.ejs": {
    "stateCleared": "modified",
  },
  "generators/entities/command.mjs": {
    "stateCleared": "modified",
  },
  "generators/entities/generator.mjs": {
    "stateCleared": "modified",
  },
  "generators/entities/generator.spec.mjs": {
    "stateCleared": "modified",
  },
  "generators/entities/index.mjs": {
    "stateCleared": "modified",
  },
  "generators/entities/templates/template-file-entities.ejs": {
    "stateCleared": "modified",
  },
  "generators/entity/command.mjs": {
    "stateCleared": "modified",
  },
  "generators/entity/generator.mjs": {
    "stateCleared": "modified",
  },
  "generators/entity/index.mjs": {
    "stateCleared": "modified",
  },
  "generators/entity/templates/template-file-entity.ejs": {
    "stateCleared": "modified",
  },
  "generators/export-jdl/command.mjs": {
    "stateCleared": "modified",
  },
  "generators/export-jdl/generator.mjs": {
    "stateCleared": "modified",
  },
  "generators/export-jdl/generator.spec.mjs": {
    "stateCleared": "modified",
  },
  "generators/export-jdl/index.mjs": {
    "stateCleared": "modified",
  },
  "generators/export-jdl/templates/template-file-export-jdl.ejs": {
    "stateCleared": "modified",
  },
  "generators/feign-client/command.mjs": {
    "stateCleared": "modified",
  },
  "generators/feign-client/generator.mjs": {
    "stateCleared": "modified",
  },
  "generators/feign-client/generator.spec.mjs": {
    "stateCleared": "modified",
  },
  "generators/feign-client/index.mjs": {
    "stateCleared": "modified",
  },
  "generators/feign-client/templates/template-file-feign-client.ejs": {
    "stateCleared": "modified",
  },
  "generators/gatling/command.mjs": {
    "stateCleared": "modified",
  },
  "generators/gatling/generator.mjs": {
    "stateCleared": "modified",
  },
  "generators/gatling/generator.spec.mjs": {
    "stateCleared": "modified",
  },
  "generators/gatling/index.mjs": {
    "stateCleared": "modified",
  },
  "generators/gatling/templates/template-file-gatling.ejs": {
    "stateCleared": "modified",
  },
  "generators/generate-blueprint/command.mjs": {
    "stateCleared": "modified",
  },
  "generators/generate-blueprint/generator.mjs": {
    "stateCleared": "modified",
  },
  "generators/generate-blueprint/generator.spec.mjs": {
    "stateCleared": "modified",
  },
  "generators/generate-blueprint/index.mjs": {
    "stateCleared": "modified",
  },
  "generators/generate-blueprint/templates/template-file-generate-blueprint.ejs": {
    "stateCleared": "modified",
  },
  "generators/git/command.mjs": {
    "stateCleared": "modified",
  },
  "generators/git/generator.mjs": {
    "stateCleared": "modified",
  },
  "generators/git/generator.spec.mjs": {
    "stateCleared": "modified",
  },
  "generators/git/index.mjs": {
    "stateCleared": "modified",
  },
  "generators/git/templates/template-file-git.ejs": {
    "stateCleared": "modified",
  },
  "generators/gradle/command.mjs": {
    "stateCleared": "modified",
  },
  "generators/gradle/generator.mjs": {
    "stateCleared": "modified",
  },
  "generators/gradle/generator.spec.mjs": {
    "stateCleared": "modified",
  },
  "generators/gradle/index.mjs": {
    "stateCleared": "modified",
  },
  "generators/gradle/templates/template-file-gradle.ejs": {
    "stateCleared": "modified",
  },
  "generators/heroku/command.mjs": {
    "stateCleared": "modified",
  },
  "generators/heroku/generator.mjs": {
    "stateCleared": "modified",
  },
  "generators/heroku/generator.spec.mjs": {
    "stateCleared": "modified",
  },
  "generators/heroku/index.mjs": {
    "stateCleared": "modified",
  },
  "generators/heroku/templates/template-file-heroku.ejs": {
    "stateCleared": "modified",
  },
  "generators/info/command.mjs": {
    "stateCleared": "modified",
  },
  "generators/info/generator.mjs": {
    "stateCleared": "modified",
  },
  "generators/info/generator.spec.mjs": {
    "stateCleared": "modified",
  },
  "generators/info/index.mjs": {
    "stateCleared": "modified",
  },
  "generators/info/templates/template-file-info.ejs": {
    "stateCleared": "modified",
  },
  "generators/init/command.mjs": {
    "stateCleared": "modified",
  },
  "generators/init/generator.mjs": {
    "stateCleared": "modified",
  },
  "generators/init/generator.spec.mjs": {
    "stateCleared": "modified",
  },
  "generators/init/index.mjs": {
    "stateCleared": "modified",
  },
  "generators/init/templates/template-file-init.ejs": {
    "stateCleared": "modified",
  },
  "generators/java/command.mjs": {
    "stateCleared": "modified",
  },
  "generators/java/generator.mjs": {
    "stateCleared": "modified",
  },
  "generators/java/generator.spec.mjs": {
    "stateCleared": "modified",
  },
  "generators/java/index.mjs": {
    "stateCleared": "modified",
  },
  "generators/java/templates/template-file-java.ejs": {
    "stateCleared": "modified",
  },
  "generators/jdl/command.mjs": {
    "stateCleared": "modified",
  },
  "generators/jdl/generator.mjs": {
    "stateCleared": "modified",
  },
  "generators/jdl/generator.spec.mjs": {
    "stateCleared": "modified",
  },
  "generators/jdl/index.mjs": {
    "stateCleared": "modified",
  },
  "generators/jdl/templates/template-file-jdl.ejs": {
    "stateCleared": "modified",
  },
  "generators/kubernetes-helm/command.mjs": {
    "stateCleared": "modified",
  },
  "generators/kubernetes-helm/generator.mjs": {
    "stateCleared": "modified",
  },
  "generators/kubernetes-helm/generator.spec.mjs": {
    "stateCleared": "modified",
  },
  "generators/kubernetes-helm/index.mjs": {
    "stateCleared": "modified",
  },
  "generators/kubernetes-helm/templates/template-file-kubernetes-helm.ejs": {
    "stateCleared": "modified",
  },
  "generators/kubernetes-knative/command.mjs": {
    "stateCleared": "modified",
  },
  "generators/kubernetes-knative/generator.mjs": {
    "stateCleared": "modified",
  },
  "generators/kubernetes-knative/generator.spec.mjs": {
    "stateCleared": "modified",
  },
  "generators/kubernetes-knative/index.mjs": {
    "stateCleared": "modified",
  },
  "generators/kubernetes-knative/templates/template-file-kubernetes-knative.ejs": {
    "stateCleared": "modified",
  },
  "generators/kubernetes/command.mjs": {
    "stateCleared": "modified",
  },
  "generators/kubernetes/generator.mjs": {
    "stateCleared": "modified",
  },
  "generators/kubernetes/generator.spec.mjs": {
    "stateCleared": "modified",
  },
  "generators/kubernetes/index.mjs": {
    "stateCleared": "modified",
  },
  "generators/kubernetes/templates/template-file-kubernetes.ejs": {
    "stateCleared": "modified",
  },
  "generators/languages/command.mjs": {
    "stateCleared": "modified",
  },
  "generators/languages/generator.mjs": {
    "stateCleared": "modified",
  },
  "generators/languages/generator.spec.mjs": {
    "stateCleared": "modified",
  },
  "generators/languages/index.mjs": {
    "stateCleared": "modified",
  },
  "generators/languages/templates/template-file-languages.ejs": {
    "stateCleared": "modified",
  },
  "generators/liquibase/command.mjs": {
    "stateCleared": "modified",
  },
  "generators/liquibase/generator.mjs": {
    "stateCleared": "modified",
  },
  "generators/liquibase/generator.spec.mjs": {
    "stateCleared": "modified",
  },
  "generators/liquibase/index.mjs": {
    "stateCleared": "modified",
  },
  "generators/liquibase/templates/template-file-liquibase.ejs": {
    "stateCleared": "modified",
  },
  "generators/maven/command.mjs": {
    "stateCleared": "modified",
  },
  "generators/maven/generator.mjs": {
    "stateCleared": "modified",
  },
  "generators/maven/generator.spec.mjs": {
    "stateCleared": "modified",
  },
  "generators/maven/index.mjs": {
    "stateCleared": "modified",
  },
  "generators/maven/templates/template-file-maven.ejs": {
    "stateCleared": "modified",
  },
  "generators/project-name/command.mjs": {
    "stateCleared": "modified",
  },
  "generators/project-name/generator.mjs": {
    "stateCleared": "modified",
  },
  "generators/project-name/generator.spec.mjs": {
    "stateCleared": "modified",
  },
  "generators/project-name/index.mjs": {
    "stateCleared": "modified",
  },
  "generators/project-name/templates/template-file-project-name.ejs": {
    "stateCleared": "modified",
  },
  "generators/react/command.mjs": {
    "stateCleared": "modified",
  },
  "generators/react/generator.mjs": {
    "stateCleared": "modified",
  },
  "generators/react/generator.spec.mjs": {
    "stateCleared": "modified",
  },
  "generators/react/index.mjs": {
    "stateCleared": "modified",
  },
  "generators/react/templates/template-file-react.ejs": {
    "stateCleared": "modified",
  },
  "generators/server/command.mjs": {
    "stateCleared": "modified",
  },
  "generators/server/generator.mjs": {
    "stateCleared": "modified",
  },
  "generators/server/generator.spec.mjs": {
    "stateCleared": "modified",
  },
  "generators/server/index.mjs": {
    "stateCleared": "modified",
  },
  "generators/server/templates/template-file-server.ejs": {
    "stateCleared": "modified",
  },
  "generators/spring-cache/command.mjs": {
    "stateCleared": "modified",
  },
  "generators/spring-cache/generator.mjs": {
    "stateCleared": "modified",
  },
  "generators/spring-cache/generator.spec.mjs": {
    "stateCleared": "modified",
  },
  "generators/spring-cache/index.mjs": {
    "stateCleared": "modified",
  },
  "generators/spring-cache/templates/template-file-spring-cache.ejs": {
    "stateCleared": "modified",
  },
  "generators/spring-cloud-stream/command.mjs": {
    "stateCleared": "modified",
  },
  "generators/spring-cloud-stream/generator.mjs": {
    "stateCleared": "modified",
  },
  "generators/spring-cloud-stream/generator.spec.mjs": {
    "stateCleared": "modified",
  },
  "generators/spring-cloud-stream/index.mjs": {
    "stateCleared": "modified",
  },
  "generators/spring-cloud-stream/templates/template-file-spring-cloud-stream.ejs": {
    "stateCleared": "modified",
  },
  "generators/spring-data-cassandra/command.mjs": {
    "stateCleared": "modified",
  },
  "generators/spring-data-cassandra/generator.mjs": {
    "stateCleared": "modified",
  },
  "generators/spring-data-cassandra/generator.spec.mjs": {
    "stateCleared": "modified",
  },
  "generators/spring-data-cassandra/index.mjs": {
    "stateCleared": "modified",
  },
  "generators/spring-data-cassandra/templates/template-file-spring-data-cassandra.ejs": {
    "stateCleared": "modified",
  },
  "generators/spring-data-couchbase/command.mjs": {
    "stateCleared": "modified",
  },
  "generators/spring-data-couchbase/generator.mjs": {
    "stateCleared": "modified",
  },
  "generators/spring-data-couchbase/generator.spec.mjs": {
    "stateCleared": "modified",
  },
  "generators/spring-data-couchbase/index.mjs": {
    "stateCleared": "modified",
  },
  "generators/spring-data-couchbase/templates/template-file-spring-data-couchbase.ejs": {
    "stateCleared": "modified",
  },
  "generators/spring-data-elasticsearch/command.mjs": {
    "stateCleared": "modified",
  },
  "generators/spring-data-elasticsearch/generator.mjs": {
    "stateCleared": "modified",
  },
  "generators/spring-data-elasticsearch/generator.spec.mjs": {
    "stateCleared": "modified",
  },
  "generators/spring-data-elasticsearch/index.mjs": {
    "stateCleared": "modified",
  },
  "generators/spring-data-elasticsearch/templates/template-file-spring-data-elasticsearch.ejs": {
    "stateCleared": "modified",
  },
  "generators/spring-data-mongodb/command.mjs": {
    "stateCleared": "modified",
  },
  "generators/spring-data-mongodb/generator.mjs": {
    "stateCleared": "modified",
  },
  "generators/spring-data-mongodb/generator.spec.mjs": {
    "stateCleared": "modified",
  },
  "generators/spring-data-mongodb/index.mjs": {
    "stateCleared": "modified",
  },
  "generators/spring-data-mongodb/templates/template-file-spring-data-mongodb.ejs": {
    "stateCleared": "modified",
  },
  "generators/spring-data-neo4j/command.mjs": {
    "stateCleared": "modified",
  },
  "generators/spring-data-neo4j/generator.mjs": {
    "stateCleared": "modified",
  },
  "generators/spring-data-neo4j/generator.spec.mjs": {
    "stateCleared": "modified",
  },
  "generators/spring-data-neo4j/index.mjs": {
    "stateCleared": "modified",
  },
  "generators/spring-data-neo4j/templates/template-file-spring-data-neo4j.ejs": {
    "stateCleared": "modified",
  },
  "generators/spring-data-relational/command.mjs": {
    "stateCleared": "modified",
  },
  "generators/spring-data-relational/generator.mjs": {
    "stateCleared": "modified",
  },
  "generators/spring-data-relational/generator.spec.mjs": {
    "stateCleared": "modified",
  },
  "generators/spring-data-relational/index.mjs": {
    "stateCleared": "modified",
  },
  "generators/spring-data-relational/templates/template-file-spring-data-relational.ejs": {
    "stateCleared": "modified",
  },
  "generators/spring-websocket/command.mjs": {
    "stateCleared": "modified",
  },
  "generators/spring-websocket/generator.mjs": {
    "stateCleared": "modified",
  },
  "generators/spring-websocket/generator.spec.mjs": {
    "stateCleared": "modified",
  },
  "generators/spring-websocket/index.mjs": {
    "stateCleared": "modified",
  },
  "generators/spring-websocket/templates/template-file-spring-websocket.ejs": {
    "stateCleared": "modified",
  },
  "generators/upgrade/command.mjs": {
    "stateCleared": "modified",
  },
  "generators/upgrade/generator.mjs": {
    "stateCleared": "modified",
  },
  "generators/upgrade/generator.spec.mjs": {
    "stateCleared": "modified",
  },
  "generators/upgrade/index.mjs": {
    "stateCleared": "modified",
  },
  "generators/upgrade/templates/template-file-upgrade.ejs": {
    "stateCleared": "modified",
  },
  "generators/vue/command.mjs": {
    "stateCleared": "modified",
  },
  "generators/vue/generator.mjs": {
    "stateCleared": "modified",
  },
  "generators/vue/generator.spec.mjs": {
    "stateCleared": "modified",
  },
  "generators/vue/index.mjs": {
    "stateCleared": "modified",
  },
  "generators/vue/templates/template-file-vue.ejs": {
    "stateCleared": "modified",
  },
  "generators/workspaces/command.mjs": {
    "stateCleared": "modified",
  },
  "generators/workspaces/generator.mjs": {
    "stateCleared": "modified",
  },
  "generators/workspaces/generator.spec.mjs": {
    "stateCleared": "modified",
  },
  "generators/workspaces/index.mjs": {
    "stateCleared": "modified",
  },
  "generators/workspaces/templates/template-file-workspaces.ejs": {
    "stateCleared": "modified",
  },
  "package.json": {
    "stateCleared": "modified",
  },
  "tsconfig.json": {
    "stateCleared": "modified",
  },
  "vitest.config.ts": {
    "stateCleared": "modified",
  },
}
`;

exports[`generator - generate-blueprint with default config should write files and match snapshot 1`] = `
{
  ".blueprint/cli/commands.mjs": {
    "stateCleared": "modified",
  },
  ".blueprint/generate-sample/command.mjs": {
    "stateCleared": "modified",
  },
  ".blueprint/generate-sample/generator.mjs": {
    "stateCleared": "modified",
  },
  ".blueprint/generate-sample/index.mjs": {
    "stateCleared": "modified",
  },
  ".blueprint/generate-sample/templates/samples/sample.jdl": {
    "stateCleared": "modified",
  },
  ".eslintrc.json": {
    "stateCleared": "modified",
  },
  ".github/workflows/generator.yml": {
    "stateCleared": "modified",
  },
  ".yo-rc.json": {
    "stateCleared": "modified",
  },
  "README.md": {
    "stateCleared": "modified",
  },
  "cli/cli.cjs": {
    "stateCleared": "modified",
  },
  "package.json": {
    "stateCleared": "modified",
  },
  "tsconfig.json": {
    "stateCleared": "modified",
  },
  "vitest.config.ts": {
    "stateCleared": "modified",
  },
}
`;
